pkg/bidstrategy/resource/capacity_available_strategy.go

```diff
@@ -30,11 +29,21 @@ func NewAvailableCapacityStrategy(ctx context.Context, params AvailableCapacityS
 
 func (s *AvailableCapacityStrategy) ShouldBidBasedOnUsage(
 	ctx context.Context, request bidstrategy.BidStrategyRequest, usage models.Resources) (bidstrategy.BidStrategyResponse, error) {
-	// skip bidding if we don't have enough capacity available
 	runningCapacity := s.runningCapacityTracker.GetAvailableCapacity(ctx)
 	enqueuedCapacity := s.enqueuedCapacityTracker.GetAvailableCapacity(ctx)
 	totalCapacity := runningCapacity.Add(enqueuedCapacity)
-	return bidstrategy.NewBidResponse(usage.LessThanEq(*totalCapacity), capacityReason, totalCapacity, usage), nil
+	if usage.LessThanEq(*totalCapacity) {
+		return bidstrategy.BidStrategyResponse{
+			ShouldBid:  true,
+			ShouldWait: false,
+			Reason:     "",
+		}, nil
+	}
+	return bidstrategy.BidStrategyResponse{
+		ShouldBid:  false,
+		ShouldWait: false,
+		Reason:     fmt.Sprintf("insufficient capacity - requested: %s, available: %s", usage.String(), totalCapacity.String()),
+	}, nil
```

Comment:
This (and the following strategy) remove the use of NewBidResponse(), but it's still widely used in the other strategies (exec etc). Is the plan to remove all of the others or is it just this one that isn't clean?

Reply:
I find the change here much easier to grok, I can glance at this code and quickly understand if we are failing or succeeding, and what the reason will be in the event we fail. Prior to this change it was not immediately clear to me what was happening here. 

There were a few more reasons for this change:
- `NewBidResponse` provides a reason for failure **and success** which which made me think it failed initially, is a reason for success even necessary? It certainly makes the job history comments verbose. 
  - Failure reason: `this node does not have capacity to run the job ({CPU: 1.200000, Memory: 79 GB, Disk: 2.8 TB, GPU: 2} requested but only {CPU: 0, Memory: 79 GB, Disk: 2.8 TB, GPU: 2} is available)`
  - Success Reason: `this no does have capacity to run the job ({CPU: 1.200000, Memory: 79 GB, Disk: 2.8 TB, GPU: 2} requested but only {CPU: 0, Memory: 79 GB, Disk: 2.8 TB, GPU: 2} is available)`

- Mixed Pointers and Value types on `models.Resources`
  - The `String()` method on `models.Resources` expects a pointer receiver. `usage` is passed by value but `totalCapacity` is a pointer. resulting in failure reasons that look like:
    - `this node does not have capacity to run the job ({CPU: 1.200000, Memory: 79 GB, Disk: 2.8 TB, GPU: 2} requested but only {%!s(float64=25) %!s(uint64=1000000000) %!s(uint64=0) %!s(uint64=0) []} is available). bid rejected`
    
Overall I'd like to refactor the rest of the bid strategies to be clearer as I find the `NewBidResponse` obtuse and prone to doing the wrong thing.

Reply:
Am fine with it being used for readability.  There looks like a few other cases where it doesn't make use of the varargs and is reasonably obvious, so maybe it's just instances like this that need changing.